### PR TITLE
feat: separate rust unit and wasm-pack tests

### LINKED ISSUE #1661: Re-separate unit and wasm-pack tests and make unit tests run in CI
addresses additional feedback from #1616

### PR DESCRIPTION
**Pull request by vacekj**
fixes #1661

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -41,6 +41,10 @@
     "test:rust": {
       "dependsOn": ["compile"],
       "inputs": ["crate/src/**", "crate/Cargo.toml", "crate/Cargo.lock", "crate/tests/**"]
+    },
+    "test:rust-unit": {
+      "dependsOn": ["compile"],
```
Comment (grod220) [inline turbo.json:46]: todo: these tests should not require compile to run. Compile step is only for those tests that require the wasm-pack wasm binaries to be built.

File: packages/wasm/package.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
     "lint:strict": "tsc --noEmit && eslint src --max-warnings 0",
     "postcompile": "touch ./wasm/.npmignore",
     "test": "vitest run",
-    "test:rust": "cd crate && wasm-pack test --headless --firefox -- --target wasm32-unknown-unknown --release --features 'mock-database'"
+    "test:rust": "cd crate && wasm-pack test --headless --firefox -- --target wasm32-unknown-unknown --release --features 'mock-database'",
+    "test:rust-unit": "cd crate && cargo test --release"
```
Comment (grod220) [inline packages/wasm/package.json:24]: issue: think rust vs rust-unit isn't the clearest differentiation. What if we did `test:wasm` vs `test:rust` (or `test:cargo`)?

**Review by grod220 — APPROVED** [review]
A few comments, feel free to merge after addressing

File: turbo.json (hunk 1 above)
Comment (vacekj) [reply turbo.json:46]: Removed the depends on compile.

File: packages/wasm/package.json (hunk 2 above)
Comment (vacekj) [reply packages/wasm/package.json:24]: thanks, good idea - went with test:wasm and test:cargo here.

**Pull request merged**